linkerd/app/inbound/src/direct.rs

```diff
@@ -25,8 +25,14 @@ pub struct RefusedNoIdentity(());
 struct RefusedNoTarget;
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub struct GatewayConnection {
-    pub target: Option<NameAddr>,
+pub enum GatewayConnection {
+    Transported(Transported),
+    Legacy(ClientInfo),
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct Transported {
```

Comment:
not _totally_ sold on this naming...all traffic is "transported". maybe `WithHeader`? not gonna block on bikeshedding this though.